packages/sdk/src/chains/evm.ts

```diff
@@ -515,102 +522,133 @@ export class EvmChain implements IChain {
 		return gas
 	}
 
-	async estimateFillOrderGas(
-		order: Order,
-		venues: {
-			v2Router: HexString
-			v2Factory: HexString
-			v3Factory: HexString
-			v3Quoter: HexString
-		},
-		fillerWalletAddress: HexString,
-		intentGatewayAddress: HexString,
-		availableAssets: {
-			name: string
-			address: HexString
-			decimals: number
-		}[],
-		universalRouterAddress: HexString,
-		postGasEstimate: bigint,
-		wrappedNativeTokenSourceChain: HexString,
-	): Promise<bigint> {
-		const nativeTokenPriceUsd = await fetchTokenUsdPrice(wrappedNativeTokenSourceChain)
-		const gasPrice = await this.publicClient.getGasPrice()
-		const gasCostInWei = postGasEstimate * gasPrice
-		const postGasEstimateUsd = (gasCostInWei * nativeTokenPriceUsd) / BigInt(10 ** 18)
-
-		const fillOptions: FillOptions = {
-			relayerFee: postGasEstimateUsd + (postGasEstimateUsd * BigInt(2)) / BigInt(100),
+	/**
+	 * Estimates the gas required to fill an order on the destination chain.
+	 * This function estimates the gas required to fill the order, including the gas required to post the request to the destination chain.
+	 *
+	 * @param order - The order to fill
+	 * @param fillerWalletAddress - The address of the wallet that will fill the order
+	 * @param destChain - The destination chain to fill the order on
+	 * @returns The estimated gas amount in fee token (DAI)
+	 */
+	async estimateFillOrder(order: Order, destChain: EvmChain): Promise<bigint> {
+		const postRequest: IPostRequest = {
+			source: order.destChain,
+			dest: order.sourceChain,
+			body: constructRedeemEscrowRequestBody(order, MOCK_ADDRESS),
+			timeoutTimestamp: 0n,
+			nonce: await this.getHostNonce(),
+			from: this.chainConfigService.getIntentGatewayAddress(order.destChain),
+			to: this.chainConfigService.getIntentGatewayAddress(order.sourceChain),
 		}
 
-		let totalEthValue = 0n
+		const postGasEstimate = await this.estimateGas(postRequest)
+		const postGasEstimateFeeToken = await this.convertGasToFeeToken(postGasEstimate, this.publicClient)
+		// 2% markup
+		const RELAYER_FEE_BPS = 200n
+		const relayerFeeInFeeToken = postGasEstimateFeeToken + (postGasEstimateFeeToken * RELAYER_FEE_BPS) / 10000n
 
-		for (const output of order.outputs) {
-			const tokenAddress = bytes32ToBytes20(output.token)
-			if (tokenAddress === ADDRESS_ZERO) {
-				totalEthValue += output.amount
-			}
+		const fillOptions: FillOptions = {
+			relayerFee: relayerFeeInFeeToken,
```

Comment:
There can be a case where the feeToken on the source has a different decimal from the one on the destination. We should scale decimals to be safe